cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/test/java/io/cdap/cdap/datapipeline/DataPipelineConnectionTest.java

```diff
@@ -675,17 +676,36 @@ private SampleResponse sampleConnection(String connection, String path, int limi
       GSON.fromJson(response.getResponseBodyAsString(), SampleResponse.class);
   }
 
-  private ConnectorDetail getConnectionSpec(String connection, String path) throws IOException {
+  private ConnectorDetail getConnectionSpec(String connection, String path, @Nullable String pluginName,
+                                            @Nullable String pluginType) throws IOException {
     String url = URLEncoder.encode(
       String.format("v1/contexts/%s/connections/%s/specification", NamespaceId.DEFAULT.getNamespace(),
                     connection), StandardCharsets.UTF_8.name());
     URL validatePipelineURL = serviceURI.resolve(url).toURL();
     HttpRequest.Builder request = HttpRequest.builder(HttpMethod.POST, validatePipelineURL)
-      .withBody(GSON.toJson(new SpecGenerationRequest(path, Collections.emptyMap())));
+      .withBody(GSON.toJson(new SpecGenerationRequest(path, Collections.emptyMap(), pluginName, pluginType)));
     HttpResponse response = executeRequest(request);
     Assert.assertEquals("Wrong answer: " + response.getResponseBodyAsString(),
                         expectedCode, response.getResponseCode());
     return expectedCode != HttpURLConnection.HTTP_OK ? null :
       GSON.fromJson(response.getResponseBodyAsString(), ConnectorDetail.class);
   }
+
+  @Test
+  public void testConnectionSpec() throws Exception {
+    File directory = TEMP_FOLDER.newFolder();
+    String conn = "test_connection2";
+    ConnectionCreationRequest creationRequest = new ConnectionCreationRequest("", new PluginInfo(
+      FileConnector.NAME, Connector.PLUGIN_TYPE, null, Collections.emptyMap(),
+      // in set up we add "-mocks" as the suffix for the artifact id
+      new ArtifactSelectorConfig("system", APP_ARTIFACT_ID.getArtifact() + "-mocks",
+                                 APP_ARTIFACT_ID.getVersion())));
+    addConnection(conn, creationRequest);
+    ConnectorDetail connectorDetail = getConnectionSpec(conn, directory.getCanonicalPath(), null, null);
+    Assert.assertTrue(connectorDetail.getRelatedPlugins().size() > 1);
+
+    connectorDetail = getConnectionSpec(conn, directory.getCanonicalPath(), "file", "batchsource");
```

Comment:
You can pass in something unrelated to the connector, and check it return an empty list